XenAdmin/Controls/Ballooning/ShinyBar.cs

```diff
@@ -222,6 +222,12 @@ private static GraphicsPath GetRoundedPath(RectangleF rect, float radius)
 
         protected void DrawGrid(Graphics g, Rectangle barArea, double bytesPerPixel, double max)
         {
+            // prevent max overflows from creating an infinite loop
+            if (max <= 0)

```

Comment:
Is this check needed? `max` is always xapi values, if it ever came here as negative, we'd rather want to know about it instead of masking it. If you must, add perhaps a Debug assertion instead?

Reply:
The check is here just because if the xapi values are wrong, XenCenter loops indefinitely. They obviously shouldn't be wrong, but it can happen (happened to me 😅, so it's possible)

Reply:
I believe it was x that overflowed rather than max, I don't see how max can have overflown, if xapi values were negative, XC would have probably crashed before getting there. I'm not against defensive programming, it's just that in this case the check might cause worse confusion.

Reply:
I see your point, it's not very clear to someone that hasn't seen this before. I'll stick with Debug.Assert then 👍